### ISSUE TITLE
关于模型训练

### ISSUE THREAD

**Issue by wanghuanniupi** [TachibanaYoshino/Remote-sensing-image-semantic-segmentation#10]
你好，我再使用您模型训练的时候，遇到总是卡在model.fit()那一步，训练开始后不打印任何信息，让我感觉像是卡了一样，请问这正常吗？

**Comment by TachibanaYoshino** [comment]
This may be a problem with your data input pipeline.

**Comment by wanghuanniupi** [comment]
打扰了，训练卡住是由于数据加载的时候迭代器解析图片格式错误，已经解决了。但是我想继续请教您一下，您的训练是在什么显卡上计算的呢，我使用3080ti 显存12G，发现GPU利用率很低，像是主要时间花在了数据搬运上。请问您有这方面的经验吗？此外，标签16通道（32 bit float）的时候，数据会扩展地很大，这一点有没有办法解决呢？

**Comment by wanghuanniupi** [comment]
此外，我想请教一下您，如果我把网络移植到pytorch框架下，是不是会提高一些训练效率呢？

**Comment by wanghuanniupi** [comment]
希望您能帮忙解答一下，多谢！

**Comment by wanghuanniupi** [comment]
> This may be a problem with your data input pipeline.

打扰了，训练卡住是由于数据加载的时候迭代器解析图片格式错误，已经解决了。但是我想继续请教您一下，您的训练是在什么显卡上计算的呢，我使用3080ti 显存12G，发现GPU利用率很低，像是主要时间花在了数据搬运上。请问您有这方面的经验吗？此外，标签16通道（32 bit float）的时候，数据会扩展地很大，这一点有没有办法解决呢？
此外，我想请教一下您，如果我把网络移植到pytorch框架下，是不是会提高一些训练效率呢？
希望您能帮忙解答一下，多谢！

**Comment by wanghuanniupi** [comment]
> This may be a problem with your data input pipeline.

打扰了，训练卡住是由于数据加载的时候迭代器解析图片格式错误，已经解决了。但是我想继续请教您一下，您的训练是在什么显卡上计算的呢，我使用3080ti 显存12G，发现GPU利用率很低，像是主要时间花在了数据搬运上。请问您有这方面的经验吗？此外，标签16通道（32 bit float）的时候，数据会扩展地很大，这一点有没有办法解决呢？

**Comment by TachibanaYoshino** [comment]
My graphics card was titan xp. You need to make the one-hot labels of all the data before training, so that the speed will be faster when the training has begun.

**Comment by wanghuanniupi** [comment]
我明白了，知道问题出在哪里了，加载数据应该主要是独热码那一块耗时过大，但是如果提前做好独热码，需要耗费大量的储存空间，这个可能需要进一步地去考虑必要性以及训练、加载数据的方式。您的回答很有帮助，非常感谢您！

**Comment by changsu777** [comment]
> > This may be a problem with your data input pipeline.这可能是您的数据输入管道存在问题。
> 
> 打扰了，训练卡住是由于数据加载的时候迭代器解析图片格式错误，已经解决了。但是我想继续请教您一下，您的训练是在什么显卡上计算的呢，我使用3080ti 显存12G，发现GPU利用率很低，像是主要时间花在了数据搬运上。请问您有这方面的经验吗？此外，标签16通道（32 bit float）的时候，数据会扩展地很大，这一点有没有办法解决呢？

你好，我也遇到了训练卡住的问题，请问这个问题是怎么解决的？

**Comment by TachibanaYoshino** [comment]
The for loop sets the label value for each pixel, which is too slow.
Consider using np.whrer() to replace the for loop to batch convert categories.

```
def get_label_from_palette(label_img, palette_file='Palette.json'):
    with open(palette_file, 'r') as fp:
        text = json.load(fp)
    label = np.zeros((label_img.shape[0], label_img.shape[1]), dtype=np.uint8)
    for i, rgb in enumerate(list(text.values())):
        label[np.where(label_img == rgb )] = i
```

**Comment by changsu777** [comment]
> The for loop sets the label value for each pixel, which is too slow. Consider using np.whrer() to replace the for loop to batch convert categories.
> 
> ```
> def get_label_from_palette(label_img, palette_file='Palette.json'):
>     with open(palette_file, 'r') as fp:
>         text = json.load(fp)
>     label = np.zeros((label_img.shape[0], label_img.shape[1]), dtype=np.uint8)
>     for i, rgb in enumerate(list(text.values())):
>         label[np.where(label_img == rgb )] = i
> ```

非常感谢您的回复，这很有用！
